graphql/batch_executor.go

```diff
@@ -335,6 +355,16 @@ func resolveObjectBatch(ctx context.Context, sources []interface{}, typ *Object,
 		}
 
 		field := typ.Fields[selection.Name]
+		if field.Batch && selection.UseBatch {
```

Comment:
Maybe add a comment "// Figure out how to resolve this field depending on the type of the resolver.

Reply:
Moved this logic into a helper and added a comment.